### PR TITLE
Updates Truffle to 5.0 and Web3 to 1.0, migrates tests

### PR DESCRIPTION
**Pull request by EnchanterIO**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests

### CODE REVIEW

File: truffle/migrations/1_initial_migration.js [hunk 1]
```diff
@@ -3,23 +3,18 @@ require('dotenv').config({ path: `${process.env.PWD}/.env` });
 const { waitForAccountToUnlock, extractEnvAccountAndPwd } = require('../test/utils');
 const Migrations = artifacts.require("./Migrations.sol");
 
-module.exports = (deployer) => {
+module.exports = async (deployer) => {
```
Comment (ggarri) [inline truffle/migrations/1_initial_migration.js]: This method cannot be `async` otherwise the main execution thread will keep going without waiting inside statements to completed

File: truffle/migrations/1_initial_migration.js [hunk 2]
```diff
@@ -3,23 +3,18 @@ require('dotenv').config({ path: `${process.env.PWD}/.env` });
...
+  let isUnlock = await web3.eth.personal.unlockAccount(from, pwd, 1000);
   if (!isUnlock) {
     console.error(`Account ${from} could not be unlock`);
     process.exit(1);
   }
 
-  // waitForAccountToUnlock(from).then(() => {
   console.log(`Account ${from} was unlocked successfully.`);
   console.log("Deploying `Migrations.sol` ...");
-  deployer.deploy(Migrations, { overwrite: false }).then(() => {
```
Comment (ggarri) [inline truffle/migrations/1_initial_migration.js]: It was useful to know when a migration deployment was completed. Why did you remove this?

File: truffle/migrations/2_deploy_hello_blockchain_world.js [hunk 3]
```diff
@@ -1,8 +1,7 @@
 const HelloBlockchainWorld = artifacts.require("./HelloBlockchainWorld.sol");
 
 module.exports = (deployer) => {
-  console.log("Deploying `HelloBlockchainWorld.sol` ...");
-  deployer.deploy(HelloBlockchainWorld, { overwrite: false }).then(() => {
```
Comment (ggarri) [inline truffle/migrations/2_deploy_hello_blockchain_world.js]: Why did you modify this? Is there any inconvenient in keeping this logs?

File: truffle/package.json [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
-  "name": "Lightchain-truffle",
-  "version": "0.0.1",
+  "name": "Lightchain-Truffle",
+  "version": "1.0.0",
   "description": "Servers as truffle suite and support for JS based parts of Lightchain codebase.",
   "scripts": {
-    "test-mainnet": "truffle test --reset --network=mainnet",
-    "test-sirius": "truffle test --reset --network=sirius",
-    "test-standalone": "truffle test --reset --network=standalone"
+    "test-mainnet": "./node_modules/.bin/truffle test --reset --network=mainnet",
```
Comment (ggarri) [inline truffle/package.json]: By default whe you run `npm` scripts this includes `./node_modules/.bin` as part of PATHs. So please, remove `./node_modules/.bin`

File: truffle/test/04_workload.js [hunk 5]
```diff
@@ -12,22 +12,20 @@ describe('Workload', async () => {
   const NEW_ACCOUNT_PASS = "password";
   
   it("should create an account for testing purposes, not asserting", async () => {
-    NEW_ACCOUNT_ADDR = await web3.personal.newAccount(NEW_ACCOUNT_PASS);
-    await web3.personal.unlockAccount(NEW_ACCOUNT_ADDR, NEW_ACCOUNT_PASS, 1000);
+    NEW_ACCOUNT_ADDR = await web3.eth.personal.newAccount(NEW_ACCOUNT_PASS);
+    await web3.eth.personal.unlockAccount(NEW_ACCOUNT_ADDR, NEW_ACCOUNT_PASS, 1000);
 
-    const txReceiptId = await web3.eth.sendTransaction({
+    await web3.eth.sendTransaction({
```
Comment (ggarri) [inline truffle/test/04_workload.js]: Did you check if this new version already fetch receipt automatically? If so, we probably should validate the status in order to not derive the assertion the following test causing a more than likely confusion on our future debugging,

File: truffle/test/04_workload.js [hunk 6]
```diff
@@ -43,31 +41,31 @@ describe('Workload', async () => {
       });
       sentFundTxReceiptPromises.push(txReceiptPromise);
     }
-    
+
     while(sentFundTxReceiptPromises.length) {
       const txReceiptId = await sentFundTxReceiptPromises.pop();
```
Comment (ggarri) [inline truffle/test/04_workload.js]: These are NOT txreceiptIds anymore, they are actual Receipts. imo you should have merge both `while` into a single one as the second one do not improve the performance of the execution and it bring a unnecesary overhead

File: truffle/migrations/1_initial_migration.js (hunk 1 above)
Comment (EnchanterIO) [reply truffle/migrations/1_initial_migration.js]: yea you are right but for some bloody reason Truffle doesn't seem to wait even if this would be a blocking method or promise. They run things in paralel but screw it, I rewrote it using promises.

Screen Shot 2019-03-18 at 09 38 59

File: truffle/migrations/1_initial_migration.js (hunk 2 above)
Comment (EnchanterIO) [reply truffle/migrations/1_initial_migration.js]: hm ok, i will add it, it wasn't necessary before as there is plenty of logs outputed already but given the circumstanes how badly Truffle runs them and all the race conditions i will put it back.

File: truffle/migrations/2_deploy_hello_blockchain_world.js (hunk 3 above)
Comment (EnchanterIO) [reply truffle/migrations/2_deploy_hello_blockchain_world.js]: answered above.

File: truffle/package.json (hunk 4 above)
Comment (EnchanterIO) [reply truffle/package.json]: oh nice didn't know, i put it back.

File: truffle/test/04_workload.js (hunk 5 above)
Comment (EnchanterIO) [reply truffle/test/04_workload.js]: yep it does, ok i will add a check as well for the status,

File: truffle/test/04_workload.js (hunk 6 above)
Comment (EnchanterIO) [reply truffle/test/04_workload.js]: yea i know, was late friday hour and it was working so I was lazy. Merged.
